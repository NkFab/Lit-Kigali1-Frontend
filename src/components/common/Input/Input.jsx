import React from 'react';
import PropTypes from 'prop-types';

const Input = ({
  type,
  classes,
  onChange,
  placeholder,
  value,
  name,
  dataTest,
}) => (
  <input
    name={name}
    type={type}
    value={value}
    className={classes}
    onChange={onChange}
    placeholder={placeholder}
    data-test={dataTest}
  />
);

Input.propTypes = {
  name: PropTypes.string,
  type: PropTypes.string,
  classes: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string,
  dataTest: PropTypes.string,
};

Input.defaultProps = {
  name: '',
  type: 'text',
  classes: 'input',
  placeholder: '',
  value: '',
  dataTest: '',
};

export default Input;
