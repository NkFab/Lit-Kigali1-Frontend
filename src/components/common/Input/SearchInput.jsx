import React, { Component } from 'react';
import PropTypes from 'prop-types';

const WAIT_INTERVAL = 1000; // In milliseconds
const ENTER_KEY = 13; // Enter keyboard press number

class SearchInput extends Component {
  state = {
    value: '',
  };

  timer = null;

  handleChange = (e) => {
    clearTimeout(this.timer);
    this.setState({ value: e.target.value });
    this.timer = setTimeout(this.triggerChange, WAIT_INTERVAL);
  };

  handleKeyDown = (e) => {
    if (e.keyCode === ENTER_KEY) {
      clearTimeout(this.timer);
      this.setState({ value: e.target.value }, () => this.triggerChange());
      e.preventDefault();
    }
  };

  triggerChange = () => {
    const { value } = this.state;
    const { onChange } = this.props;
    onChange(value);
  };

  render() {
    const { type, placeholder, classes } = this.props;
    const { value } = this.state;
    return (
      <input
        type={type}
        className={classes}
        placeholder={placeholder}
        value={value}
        onChange={this.handleChange}
        onKeyDown={this.handleKeyDown}
      />
    );
  }
}

SearchInput.propTypes = {
  type: PropTypes.string,
  classes: PropTypes.string,
  placeholder: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

SearchInput.defaultProps = {
  type: 'text',
  classes: '',
  placeholder: '',
};

export default SearchInput;
