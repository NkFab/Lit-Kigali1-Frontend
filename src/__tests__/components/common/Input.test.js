import React from 'react';
import { shallow } from 'enzyme';
import Input from '../../../components/common/Input/Input';

describe('<Input />', () => {
  test('Should render the <Input />', () => {
    const handleInput = () => true;
    const wrapper = shallow(<Input onChange={handleInput} />);
    expect(wrapper).toMatchSnapshot();
  });
});
