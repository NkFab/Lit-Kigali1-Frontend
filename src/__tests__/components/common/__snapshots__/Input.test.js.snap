// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Should render the <Input /> 1`] = `ShallowWrapper {}`;
